src/dbSta/src/dbNetwork.cc

```diff
@@ -1502,15 +1502,164 @@ ObjectId dbNetwork::id(const Net* net) const
   return 0;
 }
 
+/*
+Custom dbNetwork code.
+
+All dbNets are created in the top Instance
+by default in flat flow. So to figure out their path
+name we need to search to see the pins they are connected
+to.
+
+If we have a modnet, we construct its hierarchical name
+*/
+
+const char* dbNetwork::pathName(const Net* net) const
+{
+  // note that in flat mode, because a net is always
+  // created in the top instance the path name is just
+  // its full name, ditto hierarchical mode.
+  // For a modnet in hierarchy mode things are a bit more interesting.
+
+  dbModNet* modnet = nullptr;
+  dbNet* dnet = nullptr;
+
+  staToDb(net, dnet, modnet);
+
+  if (dnet && modnet == nullptr) {
+    return dnet->getConstName();
+  }
+
+  if (modnet) {
+    std::string modnet_name = modnet->getName();
+
+    // if a top net, don't prefix with top module name
+    dbModule* parent_module = modnet->getParent();
+    if (parent_module == block_->getTopModule()) {
+      return tmpStringCopy(modnet_name.c_str());
+    }
+
+    // accumulate a hierachical name, includes top level name
+    std::string accumulated_path_name = modnet_name;
+    std::vector<dbModule*> parent_hierarchy;
+    getParentHierarchy(parent_module, parent_hierarchy);
+    for (auto db_mod : parent_hierarchy) {
+      std::string module_name = db_mod->getName();
+      accumulated_path_name = module_name + "/" + accumulated_path_name;
+    }
+    return tmpStringCopy(accumulated_path_name.c_str());
+  }
+  return nullptr;
+}
+
+/*
+  dbNets which are connected to pins which have mod nets are
+  "boundaries".
+ */
+
+class PinModNetConnection : public PinVisitor
+{
+ public:
+  PinModNetConnection(const dbNetwork* nwk);
+  void operator()(const Pin* pin) override;
+  dbModNet* modnet_;
+
+ protected:
+  const dbNetwork* db_network_;
+};
+
+PinModNetConnection::PinModNetConnection(const dbNetwork* nwk)
+    : db_network_(nwk)
+{
+  modnet_ = nullptr;
+}
+
+void PinModNetConnection::operator()(const Pin* pin)
+{
+  dbITerm* iterm;
+  dbBTerm* bterm;
+  dbModBTerm* modbterm;
+  dbModITerm* moditerm;
+
+  db_network_->staToDb(pin, iterm, bterm, moditerm, modbterm);
+
+  if (iterm && iterm->getModNet()) {
+    modnet_ = iterm->getModNet();
+  }
+  if (bterm && bterm->getModNet()) {
+    modnet_ = bterm->getModNet();
+  }
+  if (moditerm && moditerm->getModNet()) {
+    modnet_ = moditerm->getModNet();
+  }
+  if (modbterm && modbterm->getModNet()) {
+    modnet_ = modbterm->getModNet();
+  }
```

Comment:
It seems like these are mutually exclusive - should if be an if-else sequence?

Reply:
yes, fixed.